### ISSUE TITLE
v1.15.0 seems to ignore configuration.

### ISSUE THREAD

**Issue by tedo-benchling** [gavinbunney/terraform-provider-kubectl#311]
Our releases started failing right as v1.15.0 was released.

Our setup looks like:

```
locals {
  stack_name = "XXXXX";
  kubernetes_client_auth_api_version = "client.authentication.k8s.io/v1beta1"
  service_cluster_provider_exec_args = ["token", "--region", local.region, "--cluster-id", "${local.stack_name}-service"]
  oidc_url                           = replace(data.aws_eks_cluster.service_cluster.identity[0].oidc[0].issuer, "https://", "")
}

data "aws_eks_cluster" "service_cluster" {
  name = "${local.stack_name}-service"
}

provider "kubectl" {
  host                   = data.aws_eks_cluster.service_cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.service_cluster.certificate_authority[0].data)
  load_config_file       = false
  exec {
    api_version = local.kubernetes_client_auth_api_version
    args        = local.service_cluster_provider_exec_args
    command     = "aws-iam-authenticator"
  }
}
```

With version 1.14.0 we can successfully use this with a `kubectl_manifest` object.
With version 1.15.0 we get:
`Error: failed to create kubernetes rest client for read of resource: Get "http://localhost/api?timeout=32s": EOF`

**Comment by ghostbar** [comment]
I can confirm this issue. Using `~> 1.14.0` in the version constraint is our current workaround.

**Comment by awsfanboy** [comment]
I can confirm the issue.

**Comment by gaurav04** [comment]
Same issue.

> │ Error: failed to create kubernetes rest client for read of resource: Get "http://localhost/api?timeout=32s": dial tcp [::1]:80: connect: connection refused

**Comment by mcmadhan01** [comment]
I can confirm this as well. Here is what I see in TF debug logs - no other useful traces

`provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/gavinbunney/kubectl/1.15.0/darwin_arm64/terraform-provider-kubectl_v1.15.0`

**Comment by gavinbunney** [comment]
Apologies for the bad release - I've rolled the k8s apis back to the previous version and published v1.16.0 - can you try and see if the issue is resolved now?

**Comment by tedo-benchling** [comment]
Verified it is fix, thank you!

**Comment by gavinbunney** [comment]
@tedo-benchling @kyleli666 Would you be able to verify the latest v1.17.0 release works ok for your configuration? This release has bumped the k8s api versions again following a fix done in #319, so want to be sure that was the underlying cause of your issue. Thanks!
